src/main/java/com/adobe/cq/commerce/graphql/client/impl/GraphqlClientImpl.java

```diff
@@ -86,7 +88,7 @@
 import com.google.gson.Gson;
 import com.google.gson.reflect.TypeToken;
 
-@Component(service = GraphqlClient.class)
+@Component(service = {})
```

Comment:
What is the difference here `@Component(service = {})` vs `@Component`?

Reply:
When no `service` is specified bnd will register the Component as service for each interface it implements. To not register a Component as service the `service` attribute must be explicitly declared empty.

https://docs.osgi.org/specification/osgi.cmpn/7.0.0/service.component.html#org.osgi.service.component.annotations.Component.service--